src/test/scala/DSATest.scala

```diff
@@ -0,0 +1,707 @@
+      BASILConfig(
+        loading = ILLoadingConfig(
+          inputFile = "examples/interproc_pointer_arithmetic/interproc_pointer_arithmetic.adt",
+          relfFile = "examples/interproc_pointer_arithmetic/interproc_pointer_arithmetic.relf",
+          specFile = None,
+          dumpIL = None,
+        ),
+        staticAnalysis = Some(StaticAnalysisConfig()),
+        boogieTranslation = BoogieGeneratorConfig(),
+        outputPrefix = "boogie_out",
+      )
+    )
+    val program = results.ir.program
+    val dsg = results.analysis.get.locals.get(program.procs("callee"))
+    val stack8 = dsg.adjust(dsg.find(dsg.stackMapping(8).cells(0)).getPointee)
+    val stack24 = dsg.adjust(dsg.find(dsg.stackMapping(24).cells(0)).getPointee)
+
+    // stack8 points to the formal argument object
+    assert(stack8.equals(dsg.adjust(dsg.formals(R0))))
+    assert(stack8.offset == 0)
+    // stack 24 points to the formal argument object at offset 16, instead
+    assert(stack24.equals(dsg.adjust(dsg.formals(R0)).node.get.cells(16)))
+  }
+
+
+  test("internal merge") {
+    // this is an internal merge (two cells of the same node overlap and are merged together)
+    val mem = SharedMemory("mem", 10000, 10000)
+    val locAssign1 = Assign(R6, BinaryExpr(BVADD, R0, BitVecLiteral(4, 64)), Some("00001"))
+    val locAssign2 = Assign(R7, BinaryExpr(BVADD, R0, BitVecLiteral(5, 64)), Some("00002"))
+    var program = prog(
+      proc("main",
+        block("operations",
+          locAssign1, // R6 = R0 + 4
+          locAssign2, // R7 = R0 + 5
+          MemoryAssign(mem,  R7, R1, BigEndian, 64, Some("00003")),
+          MemoryAssign(mem,  R6, R2, BigEndian, 64, Some("00004")),
+          ret
+        )
+      )
+    )
+
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    program = returnUnifier.visitProgram(program)
+    val results = RunUtils.staticAnalysis(StaticAnalysisConfig(None, None, None), IRContext(Set.empty, Set.empty, Set.empty, Map.empty, Specification(Set(), Set(), Map(), List(), List(), List(), Set()), program))
+    val dsg: DSG = results.locals.get(program.mainProcedure)
+
+    // object of formals R1 and R2 are written to overlapping fields of the same node? causing them to be merged together
+    assert(dsg.adjust(dsg.formals(R1)).equals(dsg.adjust(dsg.formals(R2))))
+
+    // R6 and R7 address the same cell
+    assert(dsg.find(dsg.varToCell(locAssign1)(R6)).cell.equals(dsg.find(dsg.varToCell(locAssign2)(R7)).cell))
+    // however, they address different internal offets in those cells
+    assert(dsg.find(dsg.varToCell(locAssign1)(R6)).internalOffset == 0)
+    assert(dsg.find(dsg.varToCell(locAssign2)(R7)).internalOffset == 1)
+    // R6 points to input R1
+    assert(dsg.adjust(dsg.varToCell(locAssign1)(R6))._pointee.isDefined)
+    assert(dsg.adjust(dsg.adjust(dsg.varToCell(locAssign1)(R6)).getPointee).equals(dsg.adjust(dsg.formals(R1))))
+
+  }
+
+  test("offsetting from middle of cell to a new cell") {
```

Comment:
Looking at the dot graph for this test, it doesn't seem like it has the node that R0/R5/R6/R7 point to (all at varying offsets) point to anything - is this intentional?

Reply:
no, this is due to an issue in the DSG::toDot.it should be resolved now.